### ISSUE TITLE
Problems with integrating bootstrap-vue

### ISSUE THREAD

**Issue by mudlee** [electron-userland/electron-webpack#302]
Hi Everyone.

I have some problems with integrating bootstrap-vue into the project.
Please see [this issue](https://github.com/bootstrap-vue/bootstrap-vue/issues/3623).

Summary: somewhy webpack handles bootstrap-vue's scss as a scoped css, but the component itself expects the opposite, so scss won't be applied on htmls, because of class hashing.

Any idea how to solve it without learning webpack A-Z?

**Comment by loopmode** [comment]
Hi
Have you tried the latest version? If I remember correctly, the issue you describe only existed in 2.7.0 and should be fixed since 2.7.1.

**Comment by mudlee** [comment]
I'm using 2.7.1 unfortunately

**Comment by loopmode** [comment]
Hm. I'm a bit confused :)
Indeed, as you can see in this issue, indeed 2.7.1 has the issue as well, and the fix was supposed to land with 2.7.2. but 1) 2.7.2 is only marked as pre-release here and 2) the commit history doesn't show any changes regarding this. The commit appears to have been released with 2.7.1 - but obviously it's not.
In any case, please have a look at this comment. It shows a way to disable CSS modules altogether: https://github.com/electron-userland/electron-webpack/issues/296#issuecomment-504652118

But please try installing `electron-webpack@2.7.2` explicitly first anyways, or `@latest`. Otherwise, the custom config should definitely help your situation.

I think I'll have some time to look into this later this weekend. Need to gain some clarity regarding the problem :)

**Comment by loopmode** [comment]
Hi @mudlee 

So I did some investigation here, and I have a couple of findings.

- installing exactly `electron-webpack@2.7.2` solves the issue of scoped CSS modules. In my example, it was all I needed to fix the missing styles.
- We don't need to a custom webpack config to setup the aliases for vue - `electron-webpack` already does that for us. 
- What's important instead is to whitelist vue. In my example I might have done too much - I added all three of `vue`, `bootstrap` and `bootstrap-vue` to the `whiteListedModules`, and maybe it wasn't necessary for all of them - but that's what got rid of the "Multiple instances of Vue detected" warning for me

I created an example repo based on the quick start example.
- The repo is https://github.com/loopmode/electron-webpack-quick-start/tree/bootstrap-vue-ts
- This is the commit that added the basic setup and config, based on reading official electron-webpack docs (I have no prior vue or typescript experience..): https://github.com/loopmode/electron-webpack-quick-start/commit/abe1d0dc89d0ff5b3734c25f4f19034242046a72
- The other commit basically just adds a Vue app, which is the bootstrap-vue forms example: https://github.com/loopmode/electron-webpack-quick-start/commit/4ee9b557c302bc1b30fad6eeb95135a9ee8dde5e

So, long story short: `electron-webpack@2.7.2` fixes your issue :)

**Comment by mudlee** [comment]
Thank you for your investigation. I’ll be at mc pc on monday and I’ll check it and get back to this issue.
Your doing an awesome job here! :)

**Comment by mudlee** [comment]
@loopmode you're my hero man. It works! Thanks for the instant and accurate help! We tried several bundlers for electron and vue, but electron-webpack is the clear winner. With this quality of support it just confirms we chose the right one.

**Comment by loopmode** [comment]
Glad I could be of help quickly, but to be honest, it was only possible because it was me who introduced the CSS modules problem in 2.7.0 in the first place :)

Anyways, I'm also glad you did choose electron-webpack, and I'd like to share an honest opinion here.
So, there will be times when support isn't that good at all. I personally only joined the project as a user. Maybe pretty much like in your case, I chose this for my team, company and project. That was in 2016 and it helped us get started in no time and successfully launch a project. Back then I contributed a thing or two that we actually needed in development (e.g HMR for CSS).
Thing is, we got the project finished in just two months, and since then, I haven't worked with electron at all. Different city, company, only webdev ever since.
However, I am so thankful for all the great work that mostly @develar has put into this, allowing us to simply use this to launch successful projects, that i decided to support electron-webpack as much as I can, whenever possible. Which is.. sometimes :)

So, effectively, be aware that there will be times with no support at all - when our actual jobs keep us just too busy. 
We just recently had a phase of roughly half a year where issues were stacking up, issues that could have easily be solved if only someone took the time.
And also, be prepared to contribute, as you might miss a feature or find a bug you can easily fix yourself much faster than "we can". My project back then actually launched based on a fork, because there was no release with the features we needed, but afterwards, we switched back to the original package from npm with no troubles, as our PRs got merged. Better late than never :)

In summary, I think picking electron-webpack is worth the occasional troubles! Good luck and much success with your project, and welcome to this little community!
